satellite/metainfo/metainfo.go

```diff
@@ -702,19 +702,38 @@ func (endpoint *Endpoint) CreateBucket(ctx context.Context, req *pb.BucketCreate
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
```

Comment:
**Rational for code Changes:**
The change here checks if already bucket exists, then update the bucket with the partnerID if one exists, otherwise add the it as a new bucket.

I chose to handle the create/update of bucket info in this same function. Otherwise it needed to add a new function to this satellite endpoint and would call for quite a number of line of new code addition in the satellite and as well as the uplink side.
adding these extra 10-15 lines of code to check if bucket already exists eliminated at least 50-75 lines of new code(from crude estimation).

Reply:
i think its better to add a separate `UpdateBucket` function

Reply:
I missed that on it and @ethanadams pointed it out. I handled it with 3 lines of code and added a test case as well in TestPartnerBucketAttrs()